packages/signal-openstack/src/error.ts

```diff
@@ -1,16 +1,56 @@
 export class SignalOpenstackError extends Error {
   constructor(message: string) {
-    super("SignalOpenstackError: " + message)
+    super(message)
     this.name = "SignalOpenstackError"
   }
 }
 
-export class SignalOpenstackApiError extends Error {
+type ErrorObject = Record<string, object> | undefined
+type SignalOpenstackApiErrorOptions = {
+  message: string
   statusCode?: number
+  errorObject?: ErrorObject
+}
+export class SignalOpenstackApiError extends Error {
+  public statusCode?: number
+  public errorObject?: ErrorObject
 
-  constructor(message: string, statusCode?: number) {
-    super("SignalOpenstackApiError: " + message)
+  constructor(params: SignalOpenstackApiErrorOptions) {
+    super(params.message)
     this.name = "SignalOpenstackApiError"
-    this.statusCode = statusCode
+    this.statusCode = params.statusCode
+    const parsedError = this.parseErrorObject(params.errorObject)
+    if (parsedError) {
+      this.message = parsedError
+    }
+  }
+
+  private parseErrorObject(object: ErrorObject): string | null {
+    if (!object) return null
+    const candidates = ["message", "description", "type", "faultstring"]
+
+    function recursiveSearch(currentObj: ErrorObject): string | null {
```

Comment:
I do not understand what this method does but I would question the recursive search method cause of the complexity lets leave it as it is and later you will explain me why you could not use zod declarations for that.